src/main/groovy/io/openliberty/tools/gradle/tasks/DevTask.groovy

```diff
@@ -0,0 +1,507 @@
+
+            // make sure server.configDirectory exists
+            initializeConfigDirectory();
+            File configDirectory = server.configDirectory;
+
+            // getOutputDir returns a string
+            File serverOutputDir = new File(getOutputDir(project));
+
+            if (serverDirectory.exists()) {
+                if (ServerStatusUtil.isServerRunning(serverInstallDir, serverOutputDir, serverName)) {
+                    throw new Exception("The server " + serverName
+                            + " is already running. Terminate all instances of the server before starting dev mode.");
+                }
+            }
+
+
+            if (resourceDirs.isEmpty()) {
+                File defaultResourceDir = new File(project.getRootDir() + "/src/main/resources");
+                logger.info("No resource directory detected, using default directory: " + defaultResourceDir);
+                resourceDirs.add(defaultResourceDir);
+            }
+
+            String artifactId = project.getName();
+
+            // create an executor for tests with an additional queue of size 1, so
+            // any further changes detected mid-test will be in the following run
+            final ThreadPoolExecutor executor = new ThreadPoolExecutor(1, 1, 0L, TimeUnit.MILLISECONDS,
+                    new ArrayBlockingQueue<Runnable>(1, true));
+
+            ProjectConnection gradleConnection = initGradleProjectConnection();
+            BuildLauncher gradleBuildLauncher = gradleConnection.newBuild();
+            try {
+                /*
+                Running the installApps task runs all tasks it depends on:
+                    :libertyStop
+                    :clean
+                    :installLiberty
+                    :libertyCreate
+                    :compileJava
+                    :processResources
+                    :classes
+                    :war
+                    :installApps
+                 */
+                runGradleTask(gradleBuildLauncher, 'libertyCreate');
+                runGradleTask(gradleBuildLauncher, 'installFeature');
+                runGradleTask(gradleBuildLauncher, 'installApps');
+            } finally {
+                gradleConnection.close();
+            }
+
+            util = new DevTaskUtil(
+                    serverDirectory, sourceDirectory, testSourceDirectory, configDirectory,
+                    resourceDirs, hotTests, skipTests, artifactId, serverStartTimeout,
+                    verifyTimeout, appUpdateTimeout, compileWait, libertyDebug
+            );
+
+//            Use the gradle compile task instead of using the DevUtil compile
+//            util.setUseMavenOrGradleCompile(true);
+
+            util.addShutdownHook(executor);
+
+            util.startServer();
+
+            List<String> artifactPaths = util.getArtifacts();
+            File buildFile = project.getBuildFile();
+            File serverXMLFile = getServerXMLFile(server);
```

Comment:
I don't think this getServerXMLFile method is really necessary. The watchFiles method accepts a null for serverXmlFile. If server.serverXmlFile is null, then watchFiles will already monitor the server.xml in the configDirectory. This is what ci.maven does.